tests/Unit/UseStatesDirectlyInEventsTest.php

```diff
@@ -101,13 +101,33 @@
     $this->assertEquals($event2->id, $user_request2->last_event_id);
 });
 
+it('supports union typed properties in events', function() {
```

Comment:
I feel like there be a second test that proves the unhappy path as well.  

Setting the value to a boolean or an array for example.

Reply:
Isn't that just going to be a PHP TypeError exception?